src/plugins/connectionPlugin.ts

```diff
@@ -353,7 +354,9 @@ export const connectionPlugin = (connectionPluginConfig?: ConnectionPluginConfig
       // field definition.
       if (pluginExtendConnection) {
         eachObj(pluginExtendConnection, (val, key) => {
-          dynamicConfig.push(`${key}: core.SubFieldResolver<TypeName, FieldName, "${key}">`)
+          dynamicConfig.push(
+            `${key}: core.FieldResolver<core.ResolverReturnTypeName<TypeName, FieldName>, "${key}">`
+          )
```

Comment:
Usage of the `ResolverReturnTypeName` typegen helper. Given the following simplified graphql schema:

```graphql
type Query {
  userConnection(...): UserConnection
}

type UserConnection {
  extendedField: String
}
```

With `TypeName = 'Query'` and `FieldName = 'userConnection'`, `ResolverReturnType<TypeName, FieldName>` will give us `'UserConnection'`.

We'll then use that to type the resolver of `extendedField` with `FieldResolver<'UserConnection', 'extendedField'>`

Reply:
(`ResolverReturnTypeName` was renamed to `FieldTypeName`)

Reply:
> (`ResolverReturnTypeName` was renamed to `FieldTypeName`)

To me these mean two different things. `ResolverReturnTypeName` would refer to the TS type while `FieldTypeName` would refer to the GrpahQL type.

Reply:
Given that the intention _is_ to refer to the GraphQL type name, I assume `FieldTypeName` is the right name